include/boost/json/detail/impl/object_impl.ipp

```diff
@@ -70,36 +70,29 @@ void
 object_impl::
 build() noexcept
 {
-    auto end = this->end();
-    for(auto p = begin(); p != end;)
+    const auto first = begin() - 1;
```

Comment:
Do we need to `BOOST_ASSERT` that the list is not empty?

Reply:
No, `first < last` is `false` if that happens.